### PR TITLE
[zig] Repackage upstream binaries

### PR DESCRIPTION
**Pull request by jeremiahpslewis**
- [x] Test local build

### CODE REVIEW

**Comment by gbaraldi** [comment]
Zig needs LLVM15 which we don't have as of now.

**Comment by jeremiahpslewis** [comment]
@gbaraldi saw that in the logs. any objections to re-packaging the binaries and then switching over to build from source when binarybuilder is at llvm15?

File: Z/zig/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    ArchiveSource("$(url_prefix)-freebsd-x86_64-$(version).tar.xz", "4e06009bd3ede34b72757eec1b5b291b30aa0d5046dadd16ecb6b34a02411254"; unpack_target = "x86_64-unknown-freebsd12.2"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd ${WORKSPACE}/srcdir/${target}/zig-*
+
+mkdir -p ${bindir}/zig/
+cp -r lib ${bindir}/zig/
+cp zig${exeext} ${bindir}/zig/
```
Comment (giordano) [inline Z/zig/build_tarballs.jl:26]: Why you're putting stuff in `zig` subdirectories?

File: Z/zig/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+chmod +x ${bindir}/*
```
Comment (giordano) [inline Z/zig/build_tarballs.jl:28]: You're making the `${bindir}/zig` directory executable?

Comment (jeremiahpslewis) [reply Z/zig/build_tarballs.jl:28]:
```suggestion
```

File: Z/zig/build_tarballs.jl (hunk 1 above)
Comment (jeremiahpslewis) [reply Z/zig/build_tarballs.jl:26]: Thanks for the review! `zig` ships with a `zig` executable and a `lib` folder which is necessary for the executable to run. At the risk of asking a terribly obvious question…where do I put the `lib` folder?

Comment (giordano) [reply Z/zig/build_tarballs.jl:26]: I guess I'm missing what's wrong with using `${libdir}`

Comment (jeremiahpslewis) [reply Z/zig/build_tarballs.jl:26]: Seems like this isn't supported. https://cvo-23052022.fly.dev/unable-to-find-zig-installation-directory-filenotfound/

``the zig binary needs to stay at the same relative path to the lib dir``

Comment (jeremiahpslewis) [reply Z/zig/build_tarballs.jl:26]: The _jll as created by the current script certainly works, but perhaps there are other issues?

**Pull request merged**